pandas/tests/frame/test_constructors.py

```diff
@@ -1762,6 +1762,39 @@ def test_constructor_datetimes_with_nulls(self, arr):
         expected = Series([np.dtype("datetime64[ns]")])
         tm.assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize("order", ["K", "A", "C", "F"])
```

Comment:
can you do the same with timedelta? not sure if that is broken as well.

Reply:
This uses numpy astype under the hood, is fine. Added tests. 

Is there a way to create a df without a numpy or pandas array which has a dtype other than timedelta64[ns]?
I casted back to compare them

Reply:
umm not sure i understand what you are asking

Reply:
Sorry, here an example:
```
expected = DataFrame(
    [
        [Timedelta(days=1), Timedelta(days=2)],
        [Timedelta(days=3), Timedelta(days=4)]
    ], dtype="timedelta64[ms]"
)
```

returns 
```
             0            1
0   86400000.0  172800000.0
1  259200000.0  345600000.0
```
with dtype float
while

```
na = np.array(
    [
        [np.timedelta64(1, 'D'), np.timedelta64(2, 'D')],
        [np.timedelta64(4, 'D'), np.timedelta64(5, 'D')]
    ],
    dtype="timedelta64[ms]",
)
df = DataFrame(na)
```
returns 
```
       0      1
0 1 days 2 days
1 4 days 5 days
```

where dtype is timedelta64[ms]

Reply:
Question is if we can create the expected in a way to have timedelta64[ms] without using numpy or pandas array

Reply:
i dont think so, no.  if it were float or int we could use a memoryview directly

Reply:
Casting to timedelta64[ns] may seem a bit odd, but does what we want here, so I would keep the current layout. Thx

Reply:
yeah agree, i think what you did is fine for the tests.